npm plugin: issues with legacyAuth

**Describe the bug**

Setting `legacyAuth` to true in `auto` config, does not properly send the value of `_auth` to `npm publish`.

**To Reproduce**

1. setup a npm project to use `auto` and the `npm` plugin
1. set `legacyAuth` to `true` in the plugin field
1. set `publishConfig.registry` in your `package.json` to be an external registry
1. make a `.env` file, add in `NPM_TOKEN`
1. run `npx auto canary` and see below error:
```
Error: Running command 'npm' with args [publish, --tag, canary, --_auth, MASKED_NPM_TOKEN] failed

npm ERR! code ERR_INVALID_ARG_TYPE
npm ERR! The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type boolean
```

(`MASKED_NPM_TOKEN` being the actual value from `.env`'s `NPM_TOKEN`)

**Expected behavior**

Should be passing the token correctly to `npm publish`.

**Additional context**

Looking at [this](https://github.com/intuit/auto/blob/master/plugins/npm/src/index.ts#L215-L218) block of code, it passes to `execPromise`, and the token is considered an argument and not the value.

If I changed those lines to the below, all is functional:
```
return [
    options.isMonorepo ? "--legacy-auth" : "--_auth" + `=${process.env.NPM_TOKEN}`,
];
```

--- a/plugins/npm/__tests__/npm-next.test.ts
+++ b/plugins/npm/__tests__/npm-next.test.ts
@@ -241,8 +241,7 @@ describe("next", () => {
       "publish",
       "--tag",
       "next",
-      "--_auth",
-      "abcd",
+      "--_auth=abcd",
     ]);
   });
 

--- a/plugins/npm/__tests__/npm.test.ts
+++ b/plugins/npm/__tests__/npm.test.ts
@@ -719,8 +719,7 @@ describe("publish", () => {
     await hooks.publish.promise({ bump: Auto.SEMVER.patch });
     expect(execPromise).toHaveBeenCalledWith("npm", [
       "publish",
-      "--_auth",
-      "abcd",
+      "--_auth=abcd",
     ]);
   });
 
@@ -996,8 +995,7 @@ describe("canary", () => {
       "publish",
       "--tag",
       "canary",
-      "--_auth",
-      "abcd",
+      "--_auth=abcd",
     ]);
   });
 

--- a/plugins/npm/src/index.ts
+++ b/plugins/npm/src/index.ts
@@ -25,15 +25,14 @@ import {
 } from "@auto-it/core";
 import getPackages from "get-monorepo-packages";
 import { gt, gte, inc, ReleaseType } from "semver";
-import { loadPackageJson, getRepo, getAuthor } from "@auto-it/package-json-utils";
+import {
+  loadPackageJson,
+  getRepo,
+  getAuthor,
+} from "@auto-it/package-json-utils";
 
 import setTokenOnCI, { getRegistry, DEFAULT_REGISTRY } from "./set-npm-token";
-import {
-  writeFile,
-  isMonorepo,
-  readFile,
-  getLernaJson,
-} from "./utils";
+import { writeFile, isMonorepo, readFile, getLernaJson } from "./utils";
 
 const { isCi } = envCi();
 const VERSION_COMMIT_MESSAGE = '"Bump version to: %s [skip ci]"';
@@ -212,10 +211,9 @@ function getLegacyAuthArgs(
     return [];
   }
 
-  return [
-    options.isMonorepo ? "--legacy-auth" : "--_auth",
-    process.env.NPM_TOKEN,
-  ];
+  return options.isMonorepo
+    ? ["--legacy-auth", process.env.NPM_TOKEN]
+    : [`--_auth=${process.env.NPM_TOKEN}`];
 }
 
 /** Get the args to set the registry. Only used with lerna */

